File: src/main/java/data/smartdeals_service/models/blog/Blog.java
package data.smartdeals_service.models.blog;

import data.smartdeals_service.models.comment.Comment;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "blogs")
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String authorName;
    @Column(nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String title;
    @Lob
    @Column(nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String content;
    @Column(nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String category;
    @Column(nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String tags;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean isPublished;
    private Integer viewCount = 0;
    private Integer likesCount = 0;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "blog")
    private List<BlogImage> thumbnailUrl;

    @OneToMany(mappedBy = "blog", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments;  // Liên kết với bảng Comment
}

File: FLUTTER/lib/features/smart_deal/data/models/blog/Blog.dart

import 'BlogImage.dart';

class Blog {
  int? id;
  String authorName;
  String title;
  String content;
  String category;
  String tags;
  DateTime createdAt;
  bool isPublished;
  List<BlogImage> thumbnailUrl;

  Blog({
    this.id,
    required this.authorName,
    required this.title,
    required this.content,
    required this.category,
    required this.tags,
    required this.createdAt,
    required this.isPublished,
    required this.thumbnailUrl,
  });

  factory Blog.fromJson(Map<String, dynamic> json) {
    return Blog(
      id: json['id'],
      authorName: json['authorName'],
      title: json['title'],
      content: json['content'],
      category: json['category'],
      tags: json['tags'],
      createdAt: DateTime(
          json['createdAt'][0],  // Năm
          json['createdAt'][1],  // Tháng
          json['createdAt'][2],  // Ngày
          json['createdAt'][3],  // Giờ
          json['createdAt'][4],  // Phút
          json['createdAt'][5],  // Giây
          (json['createdAt'][6] / 1000000).round()  // Microseconds
      ),
      isPublished: json['isPublished'],
      thumbnailUrl: (json['thumbnailUrl'] as List<dynamic>?)
          ?.map((item) => BlogImage.fromJson(item))
          .toList() ?? [],  // Nếu null, trả về danh sách rỗng
    );
  }

}
